src/prog_models/models/uav_model/uav_model.py

```diff
@@ -0,0 +1,556 @@
+            return {
+                'TrajectoryComplete': True
+            }
+
+    def simulate_to_threshold(self, future_loading_eqn, first_output = None, threshold_keys = None, **kwargs):
+        
+        if 'dt' in kwargs and kwargs['dt'] != self.parameters['dt']:
+            if isinstance(kwargs['dt'], int): 
+                warn("dt value for simulation must be equal to the constant dt value defined in model generation. To change the simulate_to time step, change the model parameter 'dt' value")
+            else: 
+                warn("dt value for simulation must be equal to the constant dt value defined in model generation. Other time step functionalities (e.g. 'auto' or function definition) are not supported with trajectory generation.")
+        kwargs['dt'] = self.parameters['dt']
+        if 'save_pts' in kwargs:
+            warn("'save_pts' functionality is not available with trajectory generation. A trajectory can only be generated for the same dt value as specified in the model generation.")
+            kwargs['save_pts'] = []
+        if 'save_freq' not in kwargs:
+            kwargs['save_freq'] = self.parameters['dt'] # if save_freq not specified, set at dt for best output 
+
+        def future_loading_new(t, x=None): 
+            if t == 0:
+                ref_now = np.concatenate((self.ref_traj.cartesian_pos[0,:], self.ref_traj.attitude[0,:], 
+                                    self.ref_traj.velocity[0,:], self.ref_traj.angular_velocity[0,:]), axis=0)
+                
+                u = self.vehicle_model.control_scheduled(self.vehicle_model.state - ref_now)
+                u[0]      += self.vehicle_model.steadystate_input
+                u[0]       = min(max([0., u[0]]), self.vehicle_model.dynamics['max_thrust'])
+     
+                return self.InputContainer({'T': u[0], 'mx': u[1], 'my': u[2], 'mz': u[3]})
+            else:
+                t_temp = np.round(t + self.parameters['dt']/2,1) 
+                time_ind = np.argmin(np.abs(t_temp - self.ref_traj.time.tolist()))
+                ref_now = np.concatenate((self.ref_traj.cartesian_pos[time_ind,:], self.ref_traj.attitude[time_ind,:], 
+                                        self.ref_traj.velocity[time_ind,:], self.ref_traj.angular_velocity[time_ind,:]), axis=0)
+
+                # Define controller
+                x_temp = np.array([x.matrix[ii][0] for ii in range(len(x.matrix)-1)])
+                u = self.vehicle_model.control_scheduled(x_temp - ref_now) 
+                u[0]      += self.vehicle_model.steadystate_input
+                u[0]       = min(max([0., u[0]]), self.vehicle_model.dynamics['max_thrust'])
+                return self.InputContainer({'T': u[0], 'mx': u[1], 'my': u[2], 'mz': u[3]})
+
+        # Simulate to threshold 
+        results = super().simulate_to_threshold(future_loading_new,first_output, threshold_keys, **kwargs)
+        return results 
+
+    def visualize_traj(self, pred):
```

Comment:
This could use some doc comments (triple quotes) after the function definition describing the function. These will then show up on the website documentation